pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -252,39 +252,112 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 	return false, nil
 }
 
+// mergeImageRegistryCertificates merges two distinct ConfigMap, both containing
+// trusted CAs for Image Registries. The first one is the default CA bundle for
+// OpenShift internal registry access, the latter is a custom config map that may
+// be configured by the user on image.config.openshift.io/cluster.
+func mergeImageRegistryCertificates(cfgCli openshiftconfigclientv1.ConfigV1Interface, cli coreclientv1.CoreV1Interface) (map[string]string, error) {
+	cas := make(map[string]string)
+
+	internalRegistryCAs, err := cli.ConfigMaps(operatorclient.ImageRegistryNamespace).Get(
+		internalRegistryCAName, metav1.GetOptions{},
+	)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
```

Comment:
if somebody delete the config map, are we going to remove it from merged configmap or keep it?

Reply:
if we don't know, it is probably event worthy

Reply:
events here in reconciliation will tend to be repeatedly fired.

Reply:
@mfojtik we are merging two config maps, **1** and **2**, into another, **3**. 

1. If only **1** is deleted, **3** will have the content of **2**.
2. If only **2** is deleted, **3** will have the content of **1**.
3. If **1** and **2** exist, **3** will have the content of both merged(user provided ConfigMap has precedence).
4. If **1** and **2** don't exist, **3** ceases to exist.